Send `UPDATED` when new annotations are made or updated

cubap opened the pull request:
This is small, but I'd like to see if it acts how you (@thehabes) are expecting. It does not fire when a form is initialized, but does when any annotations are made or changed. It fires once after all changed fields are saved and returned. It will not fire if there is an error in updating any single annotation or if none are changed.

thehabes commented:
That looks great, I had also looked at that spot and wondered if that could be what I wanted.  I would like to pull this down, include it in my local Lived Religion, and give it a shot.  I already have the script to catch a theoretical one, so it won't take too long.

thehabes commented on js/deer-record.js:
```diff
@@ -369,7 +369,10 @@ export default class DeerReport {
                 })
-            return Promise.all(annotations).then(() => entity)
+            return Promise.all(annotations).then(() => {
+                DEER.UTILS.broadcast(undefined,DEER.EVENTS.UPDATED, this.elem, entity)
```
This is throwing an undefined error.

I think you mean ` UTILS.broadcast(undefined,DEER.EVENTS.UPDATED, this.elem, entity)`

cubap replied:
oops. It ran for me, but I have a bit more around it that may have replicated UTILS.

thehabes commented:
Better.  I updated two fields and once both those fetches returned the 200, the event fired.  That is what I want.

`event.detail` does not contain the Annotations for the entity.  That means in catching the updated event, there is no way to tell which ones were updated.  It also means I could not do something like `document.getElementById("jsonView").innerHTML = event.detail`.  

I am not sure that I need to, or that I expected it to.  Does that cause any concern right now?  Did you expect that I would be able to tell what annotations were updated by catching the `DEER.EVENTS.UPDATED` event?

thehabes commented:
it is up on http://lived-religion.rerum.io/deer-lr/person.html?id=http://devstore.rerum.io/v1/id/5f22f1b5e4b04691d616915b

cubap commented:
> Better. I updated two fields and once both those fetches returned the 200, the event fired. That is what I want.
> 
> `event.detail` does not contain the Annotations for the entity. That means in catching the updated event, there is no way to tell which ones were updated. It also means I could not do something like `document.getElementById("jsonView").innerHTML = event.detail`.
> 
> I am not sure that I need to, or that I expected it to. Does that cause any concern right now? Did you expect that I would be able to tell what annotations were updated by catching the `DEER.EVENTS.UPDATED` event?

This is correct. All the annotations would be too much, since this would be used to rerender a view on a specific item and the cost of that render will be negligible once #71 is resolved.

thehabes commented:
> > Better. I updated two fields and once both those fetches returned the 200, the event fired. That is what I want.
> > `event.detail` does not contain the Annotations for the entity. That means in catching the updated event, there is no way to tell which ones were updated. It also means I could not do something like `document.getElementById("jsonView").innerHTML = event.detail`.
> > I am not sure that I need to, or that I expected it to. Does that cause any concern right now? Did you expect that I would be able to tell what annotations were updated by catching the `DEER.EVENTS.UPDATED` event?
> 
> This is correct. All the annotations would be too much, since this would be used to rerender a view on a specific item and the cost of that render will be negligible once #71 is resolved.

Then I would call this working and stable

cubap commented:
@thehabes Okay to approve?
